build/go.mod

```diff
@@ -1,8 +1,8 @@
 module github.com/mattermost/mattermost-plugin-starter-template/build
 
-go 1.21
+go 1.23.0
```

Comment:
So as per the PR description, the golang versions needs to match that of the server, but server is on `go 1.24.6`. How does `1.23.0` fix the issue and why are we not upgrading to match MM server's version? I'm not sure how `1.23.0` is selected as the version that fixes this bug. Some comments and reference in PR and issue would be helpful.